test/phabricator_tests/test_phabricator.py

```diff
@@ -0,0 +1,658 @@
+        mock_get_last_comment.return_value = mock_last_comment_object
+        mock_time_from_epoch.return_value = "mock_date"
+        fake_phab = MagicMock()
+        fake_reviews = self.mock.get_reviews()
+        mock_get_comments.return_value = "test comment"
+        mock_check_request_state.return_value = False
+        mock_author_data.return_value = ({'userName': 'mock_user', 'image': 'mock.com'}, [])
+
+        response = PhabricatorService().get_reviews(
+            phab=fake_phab,
+            reviews=fake_reviews,
+            raw_response=[],
+            host="www.google.com",
+            state_=None,
+            value=None,
+            duration=None,
+            show_last_comment=True)
+
+        mock_get_comments.assert_called_with(id=0, phab=fake_phab)
+        mock_get_last_comment(comments=mock_get_comments.return_value, phab=fake_phab,
+                              raw_response=[])
+        mock_time_from_epoch.assert_called_with('mock_date')
+        mock_check_request_state.assert_called_with('mock_date',
+                                                    None, None, None)
+        mock_has_new_comments.assert_not_called()
+        mock_author_data.assert_not_called()
+        self.assertEqual(response, [])
+
+    @patch(PATH + 'Phabricator')
+    @patch(PATH + 'PhabricatorService.generate_phids')
+    @patch(PATH + 'PhabricatorService.differential_query')
+    @patch(PATH + 'PhabricatorService.get_reviews')
+    def test_request_reviews_user_names(self, mock_get_reviews,
+                                        mock_dfferential_query,
+                                        mock_generate_phids,
+                                        mock_Phabricator):
+        """
+        Tests request_reviews with user names specified
+        """
+        fake_phab = MagicMock()
+        mock_generate_phids.return_value = ([], 'PHID-USER-xxxxxxxxxxxxxxxxxxxx')
+        mock_dfferential_query.return_value = 'mock_reviews'
+        mock_Phabricator.return_value = fake_phab
+        mock_get_reviews.return_value = '1'
+
+        response = PhabricatorService().request_reviews(
+            host="https://www.dummy.com",
+            token="dummy_token",
+            user_names=None,
+            state_=None,
+            value=None,
+            duration=None,
+            show_last_comment=None)
+
+        mock_dfferential_query.assert_called_with(
+            status='status-open',
+            responsibleUsers=[],
+            phab=fake_phab
+        )
+
+        mock_Phabricator.assert_called_with(
+            host="https://www.dummy.com/api/",
+            token="dummy_token"
+        )
+
+        self.assertEqual(['1'], response)
+
+    @patch(PATH + 'Phabricator')
+    @patch(PATH + 'PhabricatorService.generate_phids')
+    @patch(PATH + 'PhabricatorService.differential_query')
+    @patch(PATH + 'PhabricatorService.get_reviews')
+    def test_request_reviews_user_names(self, mock_get_reviews,
```

Comment:
Be carefull. You have two methods named `test_request_reviews_user_names`. I checked that they are different, but when having two with the same name, one of them is ignored.